src/App.tsx

```diff
@@ -1,9 +1,14 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import moviesFromServer from './api/movies.json';
 
 export const App: React.FC = () => {
+  const [query, setQuery] = useState('');
+  const visibleMovies = moviesFromServer
+    .filter(movie => (movie.title.toLowerCase().includes(query.toLowerCase()))
```

Comment:
split this with variables
Also, DRY, move this check if `str` includes `substr` to the separate function
<img width="750" alt="image" src="https://user-images.githubusercontent.com/91411692/235119574-9b2489b3-a424-4970-ade1-0d4638582098.png">
Trim your query